Src/Tests/Temporal.Sdk.Common.Tests/Serialization/TestUnnamedContainerPayloadConverter.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using Temporal.Api.Common.V1;
+using Temporal.Common.Payloads;
+using Temporal.Serialization;
+using Xunit;
+
+namespace Temporal.Sdk.Common.Tests.Serialization
+{
+    public class TestUnnamedContainerPayloadConverter
+    {
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_String()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            Payloads p = new Payloads();
+            Assert.False(instance.TrySerialize(String.Empty, p));
+            Assert.Empty(p.Payloads_);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Null()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            Payloads p = new Payloads();
+            Assert.False(instance.TrySerialize<string>(null, p));
+            Assert.Empty(p.Payloads_);
+        }
+
+        [Fact]
+        public void Test_UnnamedContainerPayloadConverter_TrySerialize_Unnamed_SerializedDataBacked()
+        {
+            UnnamedContainerPayloadConverter instance = new UnnamedContainerPayloadConverter();
+            instance.InitDelegates(new[] { new JsonPayloadConverter() });
+            Payloads p = new Payloads();
+            NewtonsoftJsonPayloadConverter converter = new NewtonsoftJsonPayloadConverter();
+            converter.Serialize(new SerializableClass { Name = "test", Value = 2 }, p);
+            PayloadContainers.Unnamed.SerializedDataBacked data = new PayloadContainers.Unnamed.SerializedDataBacked(p, converter);
+            Assert.True(instance.TrySerialize(data, p));
+            Assert.NotEmpty(p.Payloads_);
+            Assert.True(instance.TryDeserialize(p, out PayloadContainers.Unnamed.SerializedDataBacked cl));
+            Assert.NotNull(cl);
+            SerializableClass deserializedData = cl.GetValue<SerializableClass>(0);
+            Assert.Equal("test", deserializedData.Name);
+            Assert.Equal(2, deserializedData.Value);
```

Comment:
This is a great test case. I would like to add an additional to check that the converter embedded in the container is invoked when it should be:

first time you call `instance.TrySerialize(data, p)`, the `JsonPayloadConverter `you set in line `instance.InitDelegates(new[] { new JsonPayloadConverter() });` should be invoked. Since that's a test converter, we con modify it to add a respective query method or so.

After we have `cl`, use `instance` to serialize `cl` again. Now the `TrySerialize` method of the embedded `JsonPayloadConverter` should not be called, because the container serializer should notice the same underlying serializer and just round-trip the backing data.

We can also validate a match for `cl.SerializedData`.

Lastly construct a new `instance2` and apply it to the came `cl`. This time the underlying `JsonPayloadConverter` converter should be called again, since we are not using the same converter and cannot guarantee that using raw payload is valid.

We can also chat about that mechanism if you like. Perhaps it's too complex? Should we have comments in such a complex test?

Reply:
I modified that test case to account for most of this in a later PR, and added a comment about outstanding @ToDo.